filepicker/FilePickerDelegate.java uses or overrides a deprecated API.

**Describe the bug**
`flutter build appbundle`
produces the following message:
```
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/user1/.pub-cache/hosted/pub.dartlang.org/file_picker-5.2.0+1/android/src/main/java/com/mr/flutter/plugin/filepicker/FilePickerDelegate.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
```

**Platform**

- [x] Android
- [ ] iOS
- [ ] Web
- [ ] Desktop

**Platform OS version**
What version did it happen?
5.2.0+1

**Flutter Version details**
<details>
<summary>
$ flutter doctor -v
</summary>

```
[✓] Flutter (Channel stable, 3.3.2, on Linux Mint 20.3 5.4.0-126-generic, locale en_US.UTF-8)
    • Flutter version 3.3.2 on channel stable at /home/user1/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e3c29ec00c (12 days ago), 2022-09-14 08:46:55 -0500
    • Engine revision a4ff2c53d8
    • Dart version 2.18.1
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /home/user1/Android
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = /home/user1/Android
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2021.1)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 64.1.2
    • Dart plugin version 211.7798
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.65.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Linux Mint 20.3 5.4.0-126-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 104.0.5112.79

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

This is complete warning
D:\flutter\.pub-cache\hosted\pub.dartlang.org\file_picker-5.2.0+1\android\src\main\java\com\mr\flutter\plugin\filepicker\FilePickerDelegate.java:140: warning: [deprecation] <T>getParcelableArrayList(String) in Bundle has been deprecated
                                ArrayList<Parcelable> fileUris = bundle.getParcelableArrayList("selectedItems");
                                                                       ^
  where T is a type-variable:
    T extends Parcelable declared in method <T>getParcelableArrayList(String)
1 warning

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 5.2.2
+### Android
+- Fix deprecation warning for `getParcelable(String key)` method.
+
 ## 5.2.1
 #### Android
 - Removed build constant TIRAMISU to fix build error if targeting API 32 or less ([#1140](https://github.com/miguelpruivo/flutter_file_picker/issues/1140), [#1124](https://github.com/miguelpruivo/flutter_file_picker/issues/1124))

--- a/android/src/main/java/com/mr/flutter/plugin/filepicker/FilePickerDelegate.java
+++ b/android/src/main/java/com/mr/flutter/plugin/filepicker/FilePickerDelegate.java
@@ -137,7 +137,8 @@ public class FilePickerDelegate implements PluginRegistry.ActivityResultListener
                         } else if (data.getExtras() != null){
                             Bundle bundle = data.getExtras();
                             if (bundle.keySet().contains("selectedItems")) {
-                                ArrayList<Parcelable> fileUris = bundle.getParcelableArrayList("selectedItems");
+                                ArrayList<Parcelable> fileUris = getSelectedItems(bundle);
+
                                 int currentItem = 0;
                                 if (fileUris != null) {
                                     for (Parcelable fileUri : fileUris) {
@@ -206,6 +207,15 @@ public class FilePickerDelegate implements PluginRegistry.ActivityResultListener
 
     private static void finishWithAlreadyActiveError(final MethodChannel.Result result) {
         result.error("already_active", "File picker is already active", null);
+    }
+
+    @SuppressWarnings("deprecation")
+    private ArrayList<Parcelable> getSelectedItems(Bundle bundle){
+        if(Build.VERSION.SDK_INT >= 33){
+            return bundle.getParcelableArrayList("selectedItems", Parcelable.class);
+        }
+
+        return bundle.getParcelableArrayList("selectedItems");
     }
 
     @SuppressWarnings("deprecation")

--- a/example/android/app/build.gradle
+++ b/example/android/app/build.gradle
@@ -15,7 +15,7 @@ apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 33
 
     lintOptions {
         disable 'InvalidPackage'
@@ -24,7 +24,7 @@ android {
     defaultConfig {
         applicationId "com.mr.flutter.plugin.filepicker.example"
         minSdkVersion 16
-        targetSdkVersion 30
+        targetSdkVersion 33
         versionCode 1
         versionName "1.0"
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -3,7 +3,7 @@ description: A package that allows you to use a native file explorer to pick sin
 homepage: https://github.com/miguelpruivo/plugins_flutter_file_picker
 repository: https://github.com/miguelpruivo/flutter_file_picker
 issue_tracker: https://github.com/miguelpruivo/flutter_file_picker/issues
-version: 5.2.1
+version: 5.2.2
 
 dependencies:
   flutter:

